### PR TITLE
mobile UI mfg - Configure to be able to pre-print QR codes

### PR DESCRIPTION
**Pull request by cp-ps**

### CODE REVIEW

File: backend/de.metas.manufacturing.rest-api/src/main/java/de/metas/manufacturing/workflows_api/rest_api/json/JsonFinishGoodsReceiveQRCodesGenerateRequest.java [hunk 1]
```diff
@@ -10,27 +9,33 @@
...
 			final @NonNull FinishedGoodsReceiveLineId finishedGoodsReceiveLineId,
-			final @NonNull HuPackingInstructionsId tuPackingInstructionsId,
-			final @NonNull QtyTU qtyTUs)
+			final @NonNull HuPackingInstructionsId huPackingInstructionsId,
+			final int numberOfHUs,
+			final @Nullable Integer numberOfCopies)
 	{
-		Check.assume(qtyTUs.isPositive(), "qtyTUs is positive");
+		Check.assume(numberOfHUs > 0, "numberOfHUs is positive");
+		Check.assume(numberOfCopies == null ||  numberOfCopies >= 0, "numberOfCopies is missing or >= 0");
```
Comment (teosarca) [inline backend/de.metas.manufacturing.rest-api/src/main/java/de/metas/manufacturing/workflows_api/rest_api/json/JsonFinishGoodsReceiveQRCodesGenerateRequest.java:33]: why not >= 1?

File: misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/manufacturing/generateHUQRCodes/ConfirmOptionScreen.jsx [hunk 2]
```diff
@@ -29,17 +30,25 @@ const ConfirmOptionScreen = () => {
...
+    if (numberOfHUsEntered <= 0) {
+      return 'numberOfHUs shall be greater than 0';
+    }
+
+    return null; // OK
+  };
+
+  const validateNumberOfCopies = (numberOfCopiesEntered) => {
+    if (numberOfCopiesEntered < 0) {
+      return 'numberOfCopies shall be equal or greater than 0';
```
Comment (teosarca) [inline misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/manufacturing/generateHUQRCodes/ConfirmOptionScreen.jsx:51]: why not >= 1?

File: misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/picking/PickStepButton.jsx [hunk 3]
```diff
@@ -27,11 +29,13 @@ const PickStepButton = ({
 
   const isAlternative = !!altStepId;
   const completeStatus = computePickFromStatus(pickFrom);
+  const catchWeight = catchWeightUOM && parseQRCodeString(pickFrom.huQRCode.code)?.weightNet;
+  const captionToUse = (catchWeight ? catchWeight + catchWeightUOM : undefined) || pickFrom.locatorName;
```
Comment (teosarca) [inline misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/picking/PickStepButton.jsx:33]: why not using formatQtyToHumanReadableStr?

**Review by teosarca — CHANGES_REQUESTED** [review]
pls check my comments
and my last commits (not tested!)

File: misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/picking/PickStepButton.jsx (hunk 3 above)
Comment (cp-ps) [reply misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/picking/PickStepButton.jsx:33]: checking that out 👀

File: misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/manufacturing/generateHUQRCodes/ConfirmOptionScreen.jsx (hunk 2 above)
Comment (cp-ps) [reply misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/manufacturing/generateHUQRCodes/ConfirmOptionScreen.jsx:51]: well, I was thinking 0 copies means `"I don't want any copies"` so print the document only once :-? but tbh, after seeing ur code with that param defaulting to ONE, I think it's better to go with one here as well

**Review by teosarca — APPROVED** [review]

**Pull request merged**